src/PooledArrays.jl

```diff
@@ -307,7 +309,7 @@ Base.convert(::Type{Array}, pa::PooledArray{T, R, N}) where {T, R, N} = convert(
 function Base.getindex(pa::PooledArray, I::Integer...)
     idx = pa.refs[I...]
     iszero(idx) && throw(UndefRefError())
```

Comment:
I do not understand why we check `idx` specifically to be zero here?

Reply:
`similar` initializes refs to `zeros`

Reply:
Thank you I have missed `similar`.